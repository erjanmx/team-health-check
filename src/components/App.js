import React, { Component } from "react";
import Voter from "./Voter";
import Counter from "./Counter";

class App extends Component {
  constructor(props) {
    super(props);

    this.joinChannel = this.joinChannel.bind(this);

    this.state = {
      role: 'member',
      channelUuid: '1234',
    };
  }

  joinChannel(e) {
    e.preventDefault();

    const channelUuid = e.target.elements.channelUuid.value.trim();

    if (channelUuid) {
      this.setState({ role: 'member', channelUuid: channelUuid });
    }
  }

  render() {
    let container = 
      <form onSubmit={ this.joinChannel }>
        <h4 className="title is-4 has-text-centered">SHC</h4>
        <div className="columns">
          <div className="column">
            <div className="control">
                <input type="number" min="0" max="9999" className="input is-rounded" name="channelUuid" placeholder="Channel ID" autoComplete="off"/>
            </div>
          </div>
          <div className="column">
            <div className="control">
              <button className="button is-fullwidth is-dark is-rounded">Join</button>
            </div>
          </div>
        </div>
        <h6 className="title is-6 has-text-centered">or</h6>
        <div className="field">
          <div className="control">
            <button onClick={(e) => this.setState({ role: 'judge' })} className="button is-fullwidth is-medium">
              Create new
            </button>
          </div>
        </div>
      </form>
    
    const role = this.state.role;

    if (role === 'member') {
      container = <Voter channelUuid={this.state.channelUuid} />
    } else if (role === 'judge') {
      container = <Counter />
    }
   
    return (
      <section className="section">
        { container }
      </section>
    );
  }
}

export default App;
