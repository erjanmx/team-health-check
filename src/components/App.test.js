import React from 'react';
import ReactDOM from 'react-dom';
import { create } from "react-test-renderer";
import App from './App';

it("renders without crashing", () => {
  const div = document.createElement('div');

  ReactDOM.render(<App />, div);
});

it("is created with empty role", () => {
  const component = create(<App />);
  const instance = component.getInstance();

  expect(instance.state.role).toBe('');
});

it("it matches the snapshot of initial page", () => {
  const component = create(<App />);

  expect(component.toJSON()).toMatchSnapshot();
});
