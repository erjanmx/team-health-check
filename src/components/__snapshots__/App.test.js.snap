// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot of initial page 1`] = `
<section
  className="hero is-fullheight"
>
  <div
    className="hero-body"
  >
    <div
      className="container has-text-centered"
    >
      <div
        className="has-text-centered"
      >
        <form
          onSubmit={[Function]}
        >
          <h4
            className="title is-4"
          >
            Team health checker
          </h4>
          <hr />
          <div
            className="columns"
          >
            <div
              className="column"
            >
              <div
                className="control"
              >
                <input
                  autoComplete="off"
                  autoFocus={true}
                  className="input"
                  max="9999"
                  min="1000"
                  name="channelUuid"
                  placeholder="Channel ID"
                  type="number"
                />
              </div>
            </div>
            <div
              className="column"
            >
              <div
                className="control"
              >
                <button
                  className="button is-fullwidth is-dark"
                >
                  Join
                </button>
              </div>
            </div>
          </div>
          <h6
            className="title is-6"
          >
            or
          </h6>
          <div
            className="field"
          >
            <div
              className="control"
            >
              <button
                className="button is-fullwidth is-medium"
                onClick={[Function]}
              >
                Create
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</section>
`;
